src/seedu/addressbook/AddressBook.java

```diff
@@ -383,12 +418,175 @@ private static String executeCommand(String userInputString) {
+		return tempProperty;
+	}
+	
+    /**
+     * Checks if a user input is valid or not
+     *
+     * @param person - String userInput - what the user keyed in for the field, int - switch to the correct data validation check
+     *                 
+     * @return person - String[] representation of the new details of the person
+     */
+	private static boolean checkValidProperty(String property, int type) {
+		if (type == 0) {
+			return isPersonNameValid(property);
+		}
+		else if (type == 1) {
+			return isPersonPhoneValid(property);
+		}
+		else if (type == 2) {
+			return isPersonEmailValid(property);
+		}
+		else {
+			return isPersonSchoolValid(property);
+		}
+
+	}
+    /**
+     * looping function to find a unique person to edit, or else print an error message and take input again
+     *
+     * @param person - String commandArgs, and subsequently user input 
+     *                 
+     * @return Optional - either an empty container or a container with the person found
+     */
+    private static Optional<String[]> checkValidPersonToEdit(String commandArgs) {
+        while (true) {
+    		if (commandArgs.equals("exit")){
+    			return Optional.empty();
+    		}
+            if (commandArgs.equals("")){
+                showToUser(MESSAGE_EDIT_ERROR + " or " + MESSAGE_EDIT_EXIT);
+                executeListAllPersonsInAddressBook();
+                commandArgs = getUserInput();
+                continue;
+            }
+            final Set<String> keywords = extractKeywordsFromFindPersonArgs(commandArgs);
+            final ArrayList<String[]> personsFound = getPersonsWithNameContainingAnyKeyword(keywords);
+            if (personsFound.size() >= MAXIMUM_NUMBER_OF_PEOPLE_TO_EDIT) {
+                showToUser(MESSAGE_TOO_MANY_TO_EDIT);
+                showToUser(MESSAGE_KEY_IN_UNIQUE_PERSON + " or " + MESSAGE_EDIT_EXIT);
+                executeListAllPersonsInAddressBook();
+                commandArgs = getUserInput();
+                continue;
+            }
+            if (personsFound.size() == 0) {
+                showToUser(MESSAGE_NO_PERSONS_FOUND);
+                showToUser(MESSAGE_KEY_IN_UNIQUE_PERSON + " or " + MESSAGE_EDIT_EXIT);
+                executeListAllPersonsInAddressBook();
+                commandArgs = getUserInput();
+                continue;
+            }
+            Optional<String[]> person = Optional.of(personsFound.get(0));
+            return person;
+        }
+    }
+    /**
+     * print all the entries in the address book, sorted alphabetically
+     *
+     * @param None
+     *                 
+     * @return and prints all the people in the addressbook
+     */
+	private static String sortAddressBook() {
+		ArrayList<String[]> toBeDisplayed = getAllPersonsInAddressBook();
+		sortPersons(toBeDisplayed);
+		showToUser(toBeDisplayed);
+		return getMessageForPersonsDisplayedSummary(toBeDisplayed);
+	}
+    /**
+     * sorting function that sorts everyone in the list alphabetically
```

Comment:
comments should be in active form. better would be: `Sorts people in the address book alphabetically`